### PR TITLE
Fix package validation errors for sensor-based classes

### PR DESCRIPTION
**Pull request by mmattar**
### Proposed change(s)

Fixes all the package validation errors for sensor-based classes.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [X] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/Sensor/CameraSensor.cs [hunk 1]
```diff
@@ -58,15 +87,17 @@ public int Write(WriteAdapter adapter)
...
 
+        /// <inheritdoc/>
         public SensorCompressionType GetCompressionType()
         {
             return m_CompressionType;
         }
 
         /// <summary>
-        /// Converts a m_Camera and corresponding resolution to a 2D texture.
+        /// Converts a Camera instance and corresponding resolution to a 2D texture.
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensor/CameraSensor.cs]:
```suggestion
        /// Renders a Camera instance to a 2D texture at the corresponding resolution.
```
?

File: com.unity.ml-agents/Runtime/Sensor/CameraSensorComponent.cs [hunk 2]
```diff
@@ -1,26 +1,59 @@
-using System;
 using UnityEngine;
 
 namespace MLAgents
 {
+    /// <summary>
+    /// A SensorComponent that wraps a <see cref="CameraSensor"/>.
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensor/CameraSensorComponent.cs]:
```suggestion
    /// A SensorComponent that creates a <see cref="CameraSensor"/>.
```

File: com.unity.ml-agents/Runtime/Sensor/CameraSensor.cs [hunk 3]
```diff
@@ -102,5 +133,10 @@ public static Texture2D ObservationToTexture(Camera obsCamera, int width, int he
             RenderTexture.ReleaseTemporary(tempRt);
             return texture2D;
         }
+
+        internal static int[] GenerateShape(int width, int height, bool grayscale)
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensor/CameraSensor.cs]: Does this need a docstring?

File: com.unity.ml-agents/Runtime/Sensor/ISensor.cs [hunk 4]
```diff
@@ -14,28 +24,28 @@ public interface ISensor
     {
         /// <summary>
         /// Returns the size of the observations that will be generated.
-        /// For example, a sensor that observes the velocity of a rigid body (in 3D) would return new {3}.
-        /// A sensor that returns an RGB image would return new [] {Width, Height, 3}
+        /// For example, a sensor that observes the velocity of a rigid body (in 3D) would return
+        /// new {3}. A sensor that returns an RGB image would return new [] {Width, Height, 3}
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensor/ISensor.cs]:
```suggestion
        /// new {3}. A sensor that returns an RGB image would return new [] {Height, Width, 3}
```
oops

File: com.unity.ml-agents/Runtime/Sensor/ISensor.cs [hunk 5]
```diff
@@ -14,28 +24,28 @@ public interface ISensor
...
         int[] GetObservationShape();
 
         /// <summary>
-        /// Write the observation data directly to the WriteAdapter.
-        /// This is considered an advanced interface; for a simpler approach, use SensorBase and override WriteFloats instead.
-        /// Note that this (and GetCompressedObservation) may be called multiple times per agent step, so should not
-        /// mutate any internal state.
+        /// Write the observation data directly to the <see cref="WriteAdapter"/>.
+        /// This is considered an advanced interface; for a simpler approach, use SensorBase
+        /// and override WriteFloats instead. Note that this (and GetCompressedObservation) may
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensor/ISensor.cs]: Can we cref SensorBase and WriteFloats? and maybe GetCompressedObservation?

File: com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensor.cs [hunk 6]
```diff
@@ -308,8 +340,7 @@ public virtual SensorCompressionType GetCompressionType()
         /// Evaluates the raycasts to be used as part of an observation of an agent.
         /// </summary>
         /// <param name="input">Input defining the rays that will be cast.</param>
-        /// <param name="output">Output class that will be written to with raycast results.</param>
-        ///
+        /// <returns>Output class that will be include the raycast results.</returns>
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensor.cs]:
```suggestion
        /// <returns>Output struct containing the raycast results.</returns>
```
Open to suggestions here, but "will be include the raycast results" is bad :)

File: com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensorComponentBase.cs [hunk 7]
```diff
@@ -4,69 +4,123 @@
 
 namespace MLAgents
 {
+    /// <summary>
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensorComponentBase.cs]: Sorry, this file is going to have a lot of conflicts with my other PR.

File: com.unity.ml-agents/Runtime/Sensor/RenderTextureSensor.cs [hunk 8]
```diff
@@ -11,8 +13,15 @@ public class RenderTextureSensor : ISensor
         int[] m_Shape;
         SensorCompressionType m_CompressionType;
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensor/RenderTextureSensor.cs]: Just noticed - this should probably default to PNG

Comment (chriselion) [reply com.unity.ml-agents/Runtime/Sensor/RenderTextureSensor.cs]: Never mind, always set in the ctor

File: com.unity.ml-agents/Runtime/Sensor/ISensor.cs (hunk 5 above)
Comment (chriselion) [reply com.unity.ml-agents/Runtime/Sensor/ISensor.cs]: And actually, WriteFloats no longer exists; it's WriteObservation now. All the more reason to cref it.

File: com.unity.ml-agents/Runtime/Sensor/CameraSensor.cs (hunk 1 above)
Comment (mmattar) [reply com.unity.ml-agents/Runtime/Sensor/CameraSensor.cs]: 👍

File: com.unity.ml-agents/Runtime/Sensor/CameraSensor.cs (hunk 3 above)
Comment (mmattar) [reply com.unity.ml-agents/Runtime/Sensor/CameraSensor.cs]: Doesn't for package validation - but I added one for completeness.

File: com.unity.ml-agents/Runtime/Sensor/ISensor.cs (hunk 4 above)
Comment (mmattar) [reply com.unity.ml-agents/Runtime/Sensor/ISensor.cs]: Good catch. Did a global search and found another similar typo in Observation.cs (made the fix)

File: com.unity.ml-agents/Runtime/Sensor/ISensor.cs (hunk 5 above)
Comment (mmattar) [reply com.unity.ml-agents/Runtime/Sensor/ISensor.cs]: Done. Did a pass on the file wrt cref's.

File: com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensor.cs (hunk 6 above)
Comment (mmattar) [reply com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensor.cs]: Much better.

File: com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensorComponentBase.cs (hunk 7 above)
Comment (mmattar) [reply com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensorComponentBase.cs]: No worries.

File: com.unity.ml-agents/Runtime/Sensor/CameraSensor.cs [hunk 9]
```diff
@@ -13,28 +15,50 @@ public class CameraSensor : ISensor
         int[] m_Shape;
         SensorCompressionType m_CompressionType;
 
-        public CameraSensor(Camera camera, int width, int height, bool grayscale, string name,
-                            SensorCompressionType compression)
+        /// <summary>
+        /// Creates and returns the camera sensor.
+        /// </summary>
+        /// <param name="camera">Camera object to capture images from</param>
```
Comment (surfnerd) [inline com.unity.ml-agents/Runtime/Sensor/CameraSensor.cs]: nit: `.`'s for all of the params.

**Pull request merged**